django/contrib/postgres/fields/array.py

```diff
@@ -84,7 +84,14 @@ def db_type(self, connection):
 
     def get_db_prep_value(self, value, connection, prepared=False):
         if isinstance(value, list) or isinstance(value, tuple):
-            return [self.base_field.get_db_prep_value(i, connection, prepared=False) for i in value]
+            prepared_values = []
+            for inner_value in value:
+                if hasattr(inner_value, 'resolve_expression'):
```

Comment:
Test are passing with this if statement removed, leaving only what's in the else branch. I guess this might be an attempt to solve that tests marked as expected failures. Is it worth keeping these modifications in light of the fact that they don't work or can a test be added to show their purpose?